src/wallet/wallet.cpp

```diff
@@ -4228,8 +4228,123 @@ void CWallet::SetupDescriptorScriptPubKeyMans()
 
 void CWallet::SetActiveScriptPubKeyMan(uint256 id, OutputType type, bool internal)
 {
+    WalletLogPrintf("Setting spkMan to active: id = %s, type = %d, internal = %d\n", id.ToString(), static_cast<int>(type), static_cast<int>(internal));
     auto& spk_mans = internal ? m_internal_spk_managers : m_external_spk_managers;
     auto spk_man = m_spk_managers.at(id);
     spk_man->SetType(type, internal);
     spk_mans[type] = spk_man;
 }
+
+std::shared_ptr<DescriptorScriptPubKeyMan> CWallet::HasWalletDescriptor(const WalletDescriptor& desc) const
+{
+    if (!IsWalletFlagSet(WALLET_FLAG_DESCRIPTORS)) {
+        return nullptr;
+    }
+
+    for (auto spk_man_pair : m_spk_managers) {
+        // Try to downcast to DescriptorScriptPubKeyMan then check if the descriptors match
+        std::shared_ptr<DescriptorScriptPubKeyMan> spk_manager = std::dynamic_pointer_cast<DescriptorScriptPubKeyMan>(spk_man_pair.second);
+        if (spk_manager != nullptr && spk_manager->HasWalletDescriptor(desc)) {
+            return spk_manager;
+        }
+    }
+
+    return nullptr;
+}
+
+bool CWallet::RemoveWalletDescriptor(const WalletDescriptor& desc)
```

Comment:
I'm not sure that this is really the right way to go about reimports. Generally we have a policy of never deleting things from the wallet.